osu.Game/Screens/Play/PlayerSettings/BeatmapOffsetControl.cs

```diff
@@ -142,13 +146,16 @@ protected override void LoadComplete()
 
         private void currentChanged(ValueChangedEvent<double> offset)
         {
-            Scheduler.AddOnce(updateOffset);
```

Comment:
I forgot to highlight this when opening the PR; This is a mandatory change for automatic offset adjustment to work when the player is quick-retrying.

This is because the entire settings remain hidden during quick-retry, and when this control attempts to automatically adjust offset, the offset change is scheduled here and never gets executed until the next time a "slow" retry is performed and the settings are visible again.

Fixing this differently involves splitting this code to a separate component class that's loaded at a `PlayerLoader` level and remains updated even when settings are hidden, but I attempted that and felt I'm overengineering things.

I don't particularly see any issue with not scheduling this line, so I'm going with it.

Reply:
Your explanation above does not align.

Test case `TestAutomaticAdjustment` fails without gameplay or hidden state being involved.

Reply:
The test is actually failing due to special logic made to have the calibration button show itself when the user modifies the offset after it was automatically adjusted.

Specifically, the control adjusts the offset first then hides the button, but because the offset change callback is scheduled, the button show call executes after the hide call, and so the test fails.

This is fixed in https://github.com/ppy/osu/pull/34612/commits/5462f0c20ffb4647ccf1769edae0816e5d371ed9 by taking away non-realm-related logic from scheduling. Now scheduling doesn't break any tests.